src/Parser.php

```diff
@@ -0,0 +1,181 @@
+                continue;
+            }
+            if ($chunk == "i") {
+                $this->ansi->italic();
+                continue;
+            }
+            if ($chunk == "u") {
+                $this->ansi->underline();
+                continue;
+            }
+            if ($chunk == "bl") {
+                $this->ansi->blink();
+                continue;
+            }
+            if ($chunk == "n") {
+                $this->ansi->negative();
+                continue;
+            }
+            if ($chunk == "s") {
+                $this->ansi->strikethrough();
+                continue;
+            }
+            if (preg_match("/^f\d{1,2}/i", $chunk)) {
+                $this->ansi->color($this->fgColorLookup[substr($chunk, 1)]);
+                continue;
+            }
+            if (preg_match("/^b\d{1,2}/i", $chunk)) {
+                $this->ansi->color($this->bgColorLookup[substr($chunk, 1)]);
+                continue;
+            }
+            if ($chunk == "r") {
+                $this->ansi->nostyle();
+                continue;
+            }
+            if ($chunk == "c") {
+                $this->ansi->eraseDisplay();
+                continue;
+            }
+            if ($chunk == "eu") {
+                $this->ansi->eraseDisplayUp();
+                continue;
+            }
+            if ($chunk == "ed") {
+                $this->ansi->eraseDisplayDown();
+                continue;
+            }
+            if ($chunk == "el") {
+                $this->ansi->eraseLine();
+                continue;
+            }
+            if ($chunk == "ee") {
+                $this->ansi->eraseLineToEOL();
+                continue;
+            }
+            if ($chunk == "es") {
+                $this->ansi->eraseLineToSOL();
+                continue;
+            }
+            if (preg_match("/^cb\d{1,3}/i", $chunk)) {
+                $this->ansi->cursorBack(substr($chunk, 2));
+                continue;
+            }
+            if (preg_match("/^cf\d{1,3}/i", $chunk)) {
+                $this->ansi->cursorForward(substr($chunk, 2));
+                continue;
+            }
+            if (preg_match("/^cd\d{1,3}/i", $chunk)) {
+                $this->ansi->cursorDown(substr($chunk, 2));
+                continue;
+            }
+            if (preg_match("/^cu\d{1,3}/i", $chunk)) {
+                $this->ansi->cursorUp(substr($chunk, 2));
+                continue;
+            }
+            if (preg_match("/^xy\d{1,3},\d{1,3}/i", $chunk)) {
+                preg_match("/\d{1,3},\d{1,3}/", $chunk, $pos);
+                $pos = explode(",", $pos[0]);
+                $this->ansi->cursorPosition($pos[0], $pos[1]);
+                continue;
+            }
+            if ($chunk == "lf") {
+                $this->ansi->lf();
+                continue;
+            }
+            // finally: write
+            $this->ansi->text($chunk);
+        }
+    }
+}
```

Comment:
This file is part of your other PR and has no place here. Please remove, and PR it separately.